examples/mic_mim.py

```diff
@@ -0,0 +1,410 @@
+and the multivariate interaction measure (MIM; as well as its extension, the
+global interaction measure, GIM).
+"""
+
+# Author: Thomas S. Binns <t.s.binns@outlook.com>
+# License: BSD (3-clause)
+
+# %%
+
+import numpy as np
+from matplotlib import pyplot as plt
+from matplotlib import patheffects as pe
+
+import mne
+from mne import EvokedArray, make_fixed_length_epochs
+from mne.datasets.fieldtrip_cmc import data_path
+from mne_connectivity import seed_target_indices, spectral_connectivity_epochs
+
+###############################################################################
+# Background
+# ----------
+#
+# Multivariate forms of signal analysis allow you to simultaneously consider
+# the activity of multiple signals. In the case of connectivity, the
+# interaction between multiple sensors can be analysed at once, producing a
+# single connectivity spectrum. This approach brings not only practical
+# benefits (e.g. easier interpretability of results from the dimensionality
+# reduction), but can also offer methodological improvements (e.g. enhanced
+# signal-to-noise ratio and reduced bias).
+#
+# A popular bivariate measure of connectivity is the imaginary part of
+# coherency, which looks at the correlation between two signals in the
+# frequency domain and is immune to spurious connectivity arising from volume
+# conduction artefacts :footcite:`NolteEtAl2004`. However, depending on the
+# degree of source mixing, this measure is susceptible to biased estimates of
+# connectivity based on the spatial proximity of sensors
+# :footcite:`EwaldEtAl2012`.
+#
+# To overcome this limitation, spatial filters can be used to estimate
+# connectivity free from this source mixing-dependent bias, which additionally
+# increases the signal-to-noise ratio and allows signals to be analysed in a
+# multivariate manner :footcite:`EwaldEtAl2012`. This approach leads to the
+# following methods: the maximised imaginary part of coherency (MIC); and the
+# multivariate interaction measure (MIM).
+#
+# We start by loading some example MEG data and dividing it into
+# two-second-long epochs.
+
+# %%
+
+raw = mne.io.read_raw_ctf(data_path() / 'SubjectCMC.ds')
+raw.pick('mag')
+raw.crop(50., 110.).load_data()
+raw.notch_filter(50)
+raw.resample(100)
+
+epochs = make_fixed_length_epochs(raw, duration=2.0).load_data()
+
+###############################################################################
+# We will focus on connectivity between sensors over the left and right
+# hemispheres, with 75 sensors in the left hemisphere designated as seeds, and
+# 75 sensors in the right hemisphere designated as targets.
+
+# %%
+
+# left hemisphere sensors
+seeds = [idx for idx, ch_info in enumerate(epochs.info['chs']) if
+         ch_info['loc'][0] < 0]
+# right hemisphere sensors
+targets = [idx for idx, ch_info in enumerate(epochs.info['chs']) if
+           ch_info['loc'][0] > 0]
+
+# UNTIL RAGGED INDICES SUPPORTED
```

Comment:
```suggestion
# XXX: Currently ragged indices are not supported, so we only consider a single list of indices
```

Reply:
Is this right?

Reply:
That's right, I just reworded slightly for clarity: https://github.com/mne-tools/mne-connectivity/blob/49dd83c04c50c5dc7133e8b8ee9ccb0ff716d1a0/examples/mic_mim.py#L84-L85